python/ray/train/backend.py

```diff
@@ -66,13 +122,14 @@ class BackendExecutor:
     def __init__(
             self,
             backend_config: BackendConfig,
+            backend: Backend,
```

Comment:
nit: Undo this change since it doesn't make sense to have conflicting `BackendConfig` and `Backend`.

Reply:
Ok I made `Backend` into a singleton, so it's ok to instantiate any of the backends any number of times, and we don't need to pass around a single instance.